test/functional/test_framework/p2p.py

```diff
@@ -644,15 +644,17 @@ def test_function():
 
         self.wait_until(test_function, timeout=timeout)
 
-    def wait_for_getheaders(self, timeout=60):
-        """Waits for a getheaders message.
+    def wait_for_getheaders(self, block_hash=None, timeout=60):
+        """Waits for a getheaders message containing a specific block hash.
 
-        Receiving any getheaders message will satisfy the predicate. the last_message["getheaders"]
-        value must be explicitly cleared before calling this method, or this will return
-        immediately with success. TODO: change this method to take a hash value and only
-        return true if the correct block header has been requested."""
+        If no block hash is provided, checks whether any getheaders message has been received by the node."""
         def test_function():
-            return self.last_message.get("getheaders")
+            last_getheaders = self.last_message.get("getheaders")
```

Comment:
maybe this for https://github.com/bitcoin/bitcoin/pull/29736#discussion_r1545111761?
```suggestion
            last_getheaders = self.last_message.pop("getheaders", None)
```

Reply:
That manually check that `last_message` has a certain message after waiting fail.

What if we pass a flag (remove=False) and, if set, the element is poped, otherwise only checked?

Reply:
That manually check that `last_message` has a certain message after waiting fail.

What if we pass a flag (remove=False) and, if set, the element is poped, otherwise only checked?

Reply:
do you have any advantage/test scenario in mind where we only need to check and not pop?

i couldn't think of a situation where we still require the contents of `getheaders` message. possibilities:
1. counting bytes of message received (as in `rpc_net.py`) - but we don't need functions like `wait_for_getheaders` there.
2. even `on_getheeaders` for processing the message would happen before `wait_for_getheaders`

ideally we need to pop before and not after calling `wait_for_getheaders` function. don't think that can be done though.

(also gentle ping @theStack in case you have thoughts about this approach.)

Reply:
I don't really. I guess it depends on the expectations of what `wait_for_getheaders` is doing.

I don't think any current test fails if we change the behavior, so as long as we stick with that assumption it should be ok